jwst/associations/lib/rules_level2_base.py

```diff
@@ -560,10 +560,9 @@ def finalize(associations):
                 finalized_asns.append(asn)
 
         # Having duplicate Level 2 associations is expected.
-        # Suppress warnings.
-        with warnings.catch_warnings():
-            warnings.simplefilter('ignore')
-            lv2_asns = prune(lv2_asns)
+        #with warnings.catch_warnings():
```

Comment:
This was one of I think three instances where there was a pre-existing warning filter, but removing it did not raise any warnings in the unit tests or the regtests (this was already true before I changed pyproject.toml to blanket ignore certain warnings).  Without any warnings raised during tests, it wasn't obvious to me why this filter existed in the first place.  So, I wasn't sure what to do except remove it.

Reply:
Well, if you think it's safe to remove the warnings all together I guess we might as well remove that actual lines of code, rather than just commenting them out.

Reply:
I don't necessarily know that it is safe.  My logic is that (1) these additional warnings would be a drop in the bucket considering the amount of other RuntimeWarnings that are produced when various pieces of the code are run; (2) having no filter at all is preferable to having a blanket filter that might hide other, more impactful/meaningful warning types; (3) it feels dangerous to guess what kind of warning the original author intended to filter; (4) if these new warnings start popping up for people, it will be much easier to find calls to the relevant functions that reliably reproduce those warnings and they can be filtered then.

Let me know if you agree with that logic